bridge/client/app/_views/ktb-sequence-view/ktb-sequence-view.component.ts

```diff
@@ -104,31 +109,45 @@ export class KtbSequenceViewComponent implements OnInit, OnDestroy {
             this.project = project;
             this.updateFilterDataSource(project);
 
+            this.dataService.loadRoots(project);
+
+            this._rootsTimer = timer(0, this._tracesTimerInterval*1000)
+              .pipe(takeUntil(this.unsubscribe$))
+              .subscribe(() => {
+                // This triggers the subscription for roots$
+                this.unfinishedRoots?.forEach(root => {
+                  this.loadTraces(root);
+                })
+              });
+
             this._changeDetectorRef.markForCheck();
           });
 
-        this.dataService.roots
+        this.roots$ = this.dataService.roots
           .pipe(
             takeUntil(this.unsubscribe$),
-            filter(roots => !!roots)
-          )
-          .subscribe(roots => {
-            if (!this.currentSequence && roots && params.shkeptncontext) {
-              const root = roots.find(sequence => sequence.shkeptncontext === params.shkeptncontext);
-              let stage = params.eventId ? root.findTrace(t => t.id === params.eventId)?.getStage() : params.stage;
-              let eventId = params.eventId;
-              if (root) {
-                this.selectSequence({ root, stage, eventId });
-              } else {
-                this.dataService.loadUntilRoot(this.project, params.shkeptncontext);
+            filter(roots => !!roots),
+            tap(roots => {
+              if (!this.currentSequence && roots && params.shkeptncontext) {
+                const root = roots.find(sequence => sequence.shkeptncontext === params.shkeptncontext);
+                let stage = params.eventId ? root?.findTrace(t => t.id === params.eventId)?.getStage() : params.stage;
+                let eventId = params.eventId;
+                if (root) {
+                  this.selectSequence({ root, stage, eventId });
+                } else {
+                  this.dataService.loadUntilRoot(this.project, params.shkeptncontext);
+                }
               }
-            }
-            if (roots) {
-              this.updateFilterSequence(roots);
-              this._filterDataSource.data = this.filterFieldData;
-            }
-            this._changeDetectorRef.markForCheck();
-          });
+              if (roots) {
+                this.updateFilterSequence(roots);
+                this._filterDataSource.data = this.filterFieldData;
+                // Set unfinished roots so that the traces for updates can be loaded
+                // Also ignore currently selected root, as this is getting already polled
+                this.unfinishedRoots = roots.filter(root => !!root && root.traces.some(r => r.finished !== undefined && !r.finished)).filter(root => this.currentSequence !== root);
```

Comment:
Suggestion:
Change it to `this.unfinishedRoots = roots.filter(root => root && !root.isFinished() && root !==  this.currentSequence)`

Reply:
Changed and tested. Thanks for the input :)